numba/core/withcontexts.py

```diff
@@ -225,7 +225,10 @@ def strip_var_ver(x):
         # Verify that all outputs are annotated
         not_annotated = set(stripped_outs) - set(typeanns)
         if not_annotated:
-            msg = 'missing type annotation on outgoing variables: {}'
+            msg = (
+                'missing type annotation on outgoing variables: {0}'
+                'Example code: with objmode({0}=<add_type_here>):'
```

Comment:
I think at present the `<add_type_here>` is required to be a string spelling of the type, e.g. for an out-going `y` variable of type 1D intp array, `y='intp[:]'`.

Reply:
This seems like a weird restriction

Reply:
In hindsight it is a bit weird but made some other things easier at the time of writing, this feature is/was pretty involved. This limitation is being fixed in https://github.com/numba/numba/pull/4769.